sql/core/src/main/scala/org/apache/spark/sql/expressions/WindowSpec.scala

```diff
@@ -180,16 +180,20 @@ class WindowSpec private[sql](
   private def between(typ: FrameType, start: Long, end: Long): WindowSpec = {
     val boundaryStart = start match {
       case 0 => CurrentRow
-      case Long.MinValue => UnboundedPreceding
-      case x if x < 0 => ValuePreceding(-start.toInt)
-      case x if x > 0 => ValueFollowing(start.toInt)
+      case x if x < Int.MinValue => UnboundedPreceding
```

Comment:
shall we throw an exception if `x < Int.MinValue` and `x > Long.MinValue`? @hvanhovell  what do you think?

BTW I remember we have document to explain this behavior, we should update that too